<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Routines used in my thesis" />


<title>Luciano de Freitas Barros Neto</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">My Website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Luciano de Freitas Barros Neto</h1>
<h4 class="author">Routines used in my thesis</h4>

</div>


<p><a href="mailto:cdp_Luciano@hotmail.com"><img src="mail.png" alt="e-mail" width="30" /></a> <a href="http://lattes.cnpq.br/5456133978045879"><img src="lattes.png" alt="Lattes" width="30" /></a> <a href="https://github.com/LucianoFBNeto"><img src="github1.png" alt="GitHub" width="30" /></a> <a href="https://www.researchgate.net/profile/Luciano_Barros-Neto"><img src="RG1.png" alt="ResearchGate" width="30" /></a></p>
<div id="in-my-doctoral-thesis-i-worked-with-taxonomy-phylogeny-and-biogeography-of-a-freshwater-small-armored-catfish-of-genus-parotocinclus-in-the-northeast-region-of-brazil.-my-thesis-is-divided-into-3-chapters-and-part-of-analyzes-were-carried-out-in-r." class="section level3">
<h3>In my doctoral thesis I worked with taxonomy, phylogeny and biogeography of a freshwater small armored catfish of genus <em>Parotocinclus</em>, in the Northeast region of Brazil. My thesis is divided into 3 chapters and part of analyzes were carried out in R.</h3>
</div>
<div id="chapter-1" class="section level1">
<h1>CHAPTER 1</h1>
<div id="material-and-methods" class="section level2">
<h2>2.4. MATERIAL AND METHODS</h2>
<div id="comparative-morphology" class="section level3">
<h3>2.4.1. Comparative morphology</h3>
<p>Multivariate analysis of variance (MANOVA) with a Canonical Variate Analysis (CVA).</p>
<p>Packages used <code>ggplot2</code> (Wickham 2016), <code>ggord</code> (Beck 2017) and <code>MASS</code> (Ripley <em>et al</em>. 2020). Variance inflation factor (VIF) values were evaluated using package <code>usdm</code> (Naimi <em>et al</em>. 2014).</p>
<p>Data available on my <a href="https://github.com/LucianoFBNeto/Tese_LucianoFBNeto">GitHub</a>.</p>
<pre class="r"><code>#Open Packages
library(ggplot2)
library(ggord)
library(MASS)</code></pre>
<pre><code>## 
## Attaching package: &#39;MASS&#39;</code></pre>
<pre><code>## The following object is masked _by_ &#39;.GlobalEnv&#39;:
## 
##     genotype</code></pre>
<pre class="r"><code>library(usdm)</code></pre>
<pre><code>## Loading required package: sp</code></pre>
<pre><code>## Loading required package: raster</code></pre>
<pre><code>## 
## Attaching package: &#39;raster&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:MASS&#39;:
## 
##     area, select</code></pre>
<pre class="r"><code>#Categorizing the data
data2=data1[-51,c(-1,-2,-3,-6,-10,-13,-16,-19)]
quali=data1[-51,2]

#Viff
vif(data2) # calculates vif for the variables in r</code></pre>
<pre><code>##    Variables      VIF
## 1         BD 2.773280
## 2         HL 2.189652
## 3         TL 4.043232
## 4        AbL 2.479109
## 5         PL 2.864937
## 6        CPL 2.370739
## 7        PdL 4.614436
## 8        SNL 4.233495
## 9         OD 3.182060
## 10      AdCD 1.566629
## 11      AnCD 2.736940
## 12        DL 2.251149
## 13        MR 2.293950</code></pre>
<pre class="r"><code>vifcor(data2, th=0.9) # identify collinear variables that should be excluded</code></pre>
<pre><code>## No variable from the 13 input variables has collinearity problem. 
## 
## The linear correlation coefficients ranges between: 
## min correlation ( AdCD ~ PdL ):  0.2626471 
## max correlation ( PdL ~ TL ):  0.7729837 
## 
## ---------- VIFs of the remained variables -------- 
##    Variables      VIF
## 1         BD 2.773280
## 2         HL 2.189652
## 3         TL 4.043232
## 4        AbL 2.479109
## 5         PL 2.864937
## 6        CPL 2.370739
## 7        PdL 4.614436
## 8        SNL 4.233495
## 9         OD 3.182060
## 10      AdCD 1.566629
## 11      AnCD 2.736940
## 12        DL 2.251149
## 13        MR 2.293950</code></pre>
<pre class="r"><code>#Standardize the data
data3=scale(data2)

#Perform the CVA
cva=lda(data3, quali)

#Standard graphic of `ggplot2` package
ggord(cva, quali) </code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre class="r"><code>#Graph with shape and color adjustments in `ggord` package
p &lt;- ggord(cva, quali, poly = TRUE, ellipse=FALSE, hull=TRUE, polylntyp = quali, cols = c(&#39;#4DAF4A&#39;,&#39;#984EA3&#39;,&#39;#A65628&#39;, &#39;#4DAF4A&#39;))
p</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-4-2.png" width="672" /></p>
<pre class="r"><code>p + scale_shape_manual(&#39;Groups&#39;, values = c(23,5, 3, 2))</code></pre>
<pre><code>## Scale for &#39;shape&#39; is already present. Adding another scale for &#39;shape&#39;,
## which will replace the existing scale.</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-4-3.png" width="672" /></p>
</div>
<div id="phylogenetic-analysis" class="section level3">
<h3>2.4.2. Phylogenetic analysis</h3>
<p>For a better comparison of genes trees, the two phylogenies were plotted face to face using combined functions for analyzing and visualizing phylogenies from the R packages <code>ape</code> (Paradis, Claude &amp; Strimmer, 2004), <code>phytools</code> (Revell, 2014), <code>ggplot2</code> (Wickham, 2016), <code>ggtree</code> (Yu <em>et al</em>. 2017), <code>cowplot</code> (Wilke 2019), <code>dplyr</code> (Wickham <em>et al</em>. 2019) and <code>treeio</code> (Yu, 2019).</p>
<p>Data available on my <a href="https://github.com/LucianoFBNeto/Tese_LucianoFBNeto">GitHub</a>.</p>
<pre class="r"><code>library(ggtree)</code></pre>
<pre><code>## ggtree v1.17.4  For help: https://yulab-smu.github.io/treedata-book/
## 
## If you use ggtree in published research, please cite the most appropriate paper(s):
## 
## [36m-[39m Guangchuang Yu, Tommy Tsan-Yuk Lam, Huachen Zhu, Yi Guan. Two methods for mapping and visualizing associated data on phylogeny using ggtree. Molecular Biology and Evolution 2018, accepted. doi: 10.1093/molbev/msy194[36m-[39m Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam. ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods in Ecology and Evolution 2017, 8(1):28-36, doi:10.1111/2041-210X.12628</code></pre>
<pre><code>## 
## Attaching package: &#39;ggtree&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:raster&#39;:
## 
##     flip, rotate</code></pre>
<pre class="r"><code>library(ggplot2)
library(dplyr)</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:raster&#39;:
## 
##     intersect, select, union</code></pre>
<pre><code>## The following object is masked from &#39;package:MASS&#39;:
## 
##     select</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(cowplot)</code></pre>
<pre><code>## 
## ********************************************************</code></pre>
<pre><code>## Note: As of version 1.0.0, cowplot does not change the</code></pre>
<pre><code>##   default ggplot2 theme anymore. To recover the previous</code></pre>
<pre><code>##   behavior, execute:
##   theme_set(theme_cowplot())</code></pre>
<pre><code>## ********************************************************</code></pre>
<pre class="r"><code>library(treeio)</code></pre>
<pre><code>## 
## Attaching package: &#39;treeio&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:raster&#39;:
## 
##     mask</code></pre>
<pre class="r"><code>#Open data
setwd(&quot;C:/Users/Luciano/Desktop/R_MarkDown&quot;)
x &lt;- read.tree(&quot;CYTBBAYES.tree&quot;)
y &lt;- read.tree(&quot;COIBAYES.tree&quot;)
info &lt;- read.csv(&quot;basin.csv&quot;)

x &lt;- read.beast(&quot;CYTBBAYES.tre&quot;)


y &lt;- read.beast(&quot;COIBAYES.tre&quot;)</code></pre>
<pre class="r"><code># Phylogenetic tree 1 -----------------------------------------------------
p1 &lt;- ggtree(x) %&lt;+% info + xlim(0, 40)
p2 &lt;- p1 + geom_tiplab(offset = 2.5, hjust = .5) +
  geom_tippoint(aes(shape = Species_group, color = Basin, size = 5)) +
  scale_shape_manual(values=c(18, 5, 3, 6, 8, 1, 2, 0)) +
  scale_color_manual(values=c(&#39;#984EA3&#39;,&#39;black&#39;, &#39;#377EB8&#39;,&#39;#A65628&#39;,
                              &#39;#4DAF4A&#39;, &#39;#A6CEE3&#39;)) +
  theme(legend.position = &quot;right&quot;) +
  theme_tree2()

p3 = revts(p2)
p4 = p3 + scale_x_continuous()</code></pre>
<pre><code>## Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
## will replace the existing scale.</code></pre>
<pre class="r"><code>plot(p4)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<pre class="r"><code># Phylogenetic tree 2 -----------------------------------------------------

p5 &lt;- ggtree(y) %&lt;+% info + xlim(0, 40)
p6 &lt;- p5 + geom_tiplab(offset = 2.5, hjust = .5) +
  geom_tippoint(aes(shape = Species_group, color = Basin, size = 5)) +
  scale_shape_manual(values=c(18, 5, 3, 6, 8, 1, 2, 0)) +
  scale_color_manual(values=c(&#39;#984EA3&#39;,&#39;black&#39;, &#39;#377EB8&#39;,&#39;#A65628&#39;,
                              &#39;#4DAF4A&#39;, &#39;#A6CEE3&#39;)) +
  theme(legend.position = &quot;right&quot;) +
  theme_tree2() #lembrar de remover esse ultimo comando no final#

p7 &lt;- flip(p6, node1=13, node2=14)
plot(p7)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<pre class="r"><code>p7 = revts(p7)
p7 = p7 + scale_x_continuous()</code></pre>
<pre><code>## Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
## will replace the existing scale.</code></pre>
<pre class="r"><code>plot(p7)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-9-2.png" width="672" /></p>
<pre class="r"><code>p7&lt;-p7+theme(legend.position=&quot;right&quot;)
plot(p7)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-9-3.png" width="672" /></p>
<pre class="r"><code># plot the 2 trees side by side --------------------------------------------
cowplot::plot_grid(p4, p7, ncol=2, labels = c(&#39;CytB&#39;,&#39;Cox1&#39;))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<pre class="r"><code># Final adjustment on trees -----------------------------------------------------
d1 &lt;- p4$data
d2 &lt;- p7$data

d2$x &lt;- max(d2$x) - d2$x + max(d1$x) + 9

pp &lt;- p4  + geom_tree(data=d2) + geom_tiplab(data = d2, hjust=2)

dd &lt;- bind_rows(d1, d2) %&gt;% 
  filter(!is.na(label))

pp + geom_line(aes(x, y, group=label), data=dd, color=&#39;grey&#39;)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="chapter-2" class="section level1">
<h1>CHAPTER 2</h1>
<div id="material-and-methods-1" class="section level2">
<h2>3.4. MATERIAL AND METHODS</h2>
<div id="phylogenetic-analyzes-and-molecular-clock" class="section level3">
<h3>3.4.5. Phylogenetic analyzes and molecular clock</h3>
<p>Morphometric and meristic data plotted with a heat map associated with molecular data using combined functions to analyze and visualize phylogenies using the packages <code>ape</code> (Paradis <em>et al</em>. 2004), <code>phytools</code> (Revell 2014), <code>treeio</code> (Yu 2019), <code>ggtree</code> (Yu <em>et al</em>. 2017), <code>ggplot2</code> (Wickham 2016) and <code>ggnewscale</code> (Campitelli 2016).</p>
<p>Data available on my <a href="https://github.com/LucianoFBNeto/Tese_LucianoFBNeto">GitHub</a>.</p>
<pre class="r"><code>library(&quot;ape&quot;)</code></pre>
<pre><code>## 
## Attaching package: &#39;ape&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:treeio&#39;:
## 
##     drop.tip</code></pre>
<pre><code>## The following object is masked from &#39;package:ggtree&#39;:
## 
##     rotate</code></pre>
<pre><code>## The following objects are masked from &#39;package:raster&#39;:
## 
##     rotate, zoom</code></pre>
<pre class="r"><code>library(&quot;ggtree&quot;)
library(&quot;ggplot2&quot;)
library(&quot;ggnewscale&quot;)
library(&#39;phytools&#39;)</code></pre>
<pre><code>## Loading required package: maps</code></pre>
<pre><code>## 
## Attaching package: &#39;phytools&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:treeio&#39;:
## 
##     read.newick</code></pre>
<pre class="r"><code>library(&#39;treeio&#39;)</code></pre>
<pre class="r"><code>#Opening and Categorizing the data
setwd(&quot;C:/Users/Luciano/Desktop/R_MarkDown&quot;)
beast_tree &lt;- read.beast(&quot;Morfo_filo.tree&quot;)
genotype &lt;- read.table(&quot;Data2.txt&quot;, header=TRUE, row.names = 1)
df &lt;- genotype[,1:2]
df2 &lt;- as.data.frame(scale(genotype[,3:15]))</code></pre>
<pre class="r"><code>#Plot tree
p &lt;- ggtree(beast_tree, aes(x=&#39;Node_ages&#39;)) + theme_tree2() + scale_x_continuous(labels = abs)
p1 &lt;- revts(p)
plot(p1)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<pre class="r"><code>#Plot tips names
p2 &lt;- p1 + geom_tiplab(size=4, align=TRUE)</code></pre>
<pre><code>## Warning: Duplicated aesthetics after name standardisation: size</code></pre>
<pre class="r"><code>plot(p2)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<pre class="r"><code>#Creating heatmap with discrete data associated with the tree
p3 &lt;- (p2 + scale_y_continuous(expand=c(0, 0.3))) %&gt;%
  gheatmap(df, offset=9, width=0.1, colnames=FALSE)+
  scale_fill_viridis_d(option=&quot;D&quot;, name=&quot;discrete\nvalue&quot;) </code></pre>
<pre><code>## Warning: attributes are not identical across measure variables;
## they will be dropped</code></pre>
<pre><code>## Scale for &#39;y&#39; is already present. Adding another scale for &#39;y&#39;, which
## will replace the existing scale.</code></pre>
<pre><code>## Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;,
## which will replace the existing scale.</code></pre>
<pre class="r"><code>#Creating heatmap with continuous data associated with the tree
p4 &lt;- p3 + new_scale_fill()
gheatmap(p4, df2, offset=13.5, width=0.8,
         colnames=FALSE) +
  scale_fill_viridis_c(option=&quot;B&quot;, name=&quot;continuous\nvalue&quot;) + scale_x_ggtree()</code></pre>
<pre><code>## Scale for &#39;y&#39; is already present. Adding another scale for &#39;y&#39;, which
## will replace the existing scale.</code></pre>
<pre><code>## Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;,
## which will replace the existing scale.</code></pre>
<pre><code>## Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
## will replace the existing scale.</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="chapter-3" class="section level1">
<h1>CHAPTER 3</h1>
<div id="material-and-methods-2" class="section level2">
<h2>4.4. MATERIAL AND METHODS</h2>
<div id="morphological-analysis" class="section level3">
<h3>4.4.1. Morphological analysis</h3>
<p>Multivariate analysis of variance (MANOVA) with a Canonical Variate Analysis (CVA).</p>
<p>Packages used <code>ggplot2</code> (Wickham 2016), <code>ggord</code> (Beck 2017) and <code>MASS</code> (Ripley <em>et al</em>. 2020). Variance inflation factor (VIF) values were evaluated using package <code>usdm</code> (Naimi <em>et al</em>. 2014).</p>
<p>Data available on my <a href="https://github.com/LucianoFBNeto/Tese_LucianoFBNeto">GitHub</a>.</p>
<pre class="r"><code>#Open Packages
library(ggplot2)
library(ggord)
library(MASS)
library(usdm)</code></pre>
<pre class="r"><code>#Categorizing the data
data2=data1[,4:21]
quali=data1[,3]

#Viff
vif(data2) # calculates vif for the variables in r</code></pre>
<pre><code>##    Variables       VIF
## 1         BD 11.423268
## 2         HL 48.421409
## 3         CW 11.975266
## 4         TL  5.388162
## 5        AbL 10.949612
## 6         PL  4.110722
## 7        CPD 10.331358
## 8        CPL 14.919976
## 9        PdL 43.614464
## 10        HD 14.493862
## 11       SNL 29.125725
## 12        OD  1.935570
## 13        IW  8.574534
## 14      AdCD  3.280936
## 15      AnCD  2.480605
## 16      PAdD 45.093350
## 17        DL  4.545348
## 18        MR  3.517378</code></pre>
<pre class="r"><code>vifcor(data2, th=0.9) # identify collinear variables that should be excluded</code></pre>
<pre><code>## 5 variables from the 18 input variables have collinearity problem: 
##  
## PAdD HL PdL HD SNL 
## 
## After excluding the collinear variables, the linear correlation coefficients ranges between: 
## min correlation ( MR ~ PL ):  0.06207773 
## max correlation ( CPL ~ AbL ):  0.898937 
## 
## ---------- VIFs of the remained variables -------- 
##    Variables       VIF
## 1         BD  6.293153
## 2         CW  9.375385
## 3         TL  3.572072
## 4        AbL  9.507596
## 5         PL  3.101434
## 6        CPD  8.687367
## 7        CPL 11.593698
## 8         OD  1.497655
## 9         IW  6.530378
## 10      AdCD  2.881833
## 11      AnCD  2.391239
## 12        DL  3.861501
## 13        MR  2.663421</code></pre>
<pre class="r"><code>#Standardize the data
data3=scale(data2)

#Perform the CVA
cva=lda(data3, quali)

#Standard graphic of `ggplot2` package
ggord(cva, quali) </code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<pre class="r"><code>#Graph with shape and color adjustments in `ggord` package
p &lt;- ggord(cva, quali, poly = TRUE, ellipse=FALSE, hull=TRUE, polylntyp = quali, cols = c(&quot;#FF7F00&quot;,&quot;#1F78B4&quot;, &quot;#B15928&quot;,&quot;#33A02C&quot;,&quot;#A6CEE3&quot;,&quot;#E31A1C&quot;), size=3)
p + scale_shape_manual(&#39;Groups&#39;, values = c(25, 25, 25,6,25,25))</code></pre>
<pre><code>## Scale for &#39;shape&#39; is already present. Adding another scale for &#39;shape&#39;,
## which will replace the existing scale.</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-22-2.png" width="672" /></p>
</div>
<div id="phylogenetic-analyzes-and-molecular-clock-1" class="section level3">
<h3>4.4.5. Phylogenetic analyzes and molecular clock</h3>
<p>Morphometric and meristic data plotted with a heat map associated with molecular data using combined functions to analyze and visualize phylogenies using the packages <code>ape</code> (Paradis <em>et al</em>. 2004), <code>phytools</code> (Revell 2014), <code>treeio</code> (Yu 2019), <code>ggtree</code> (Yu <em>et al</em>. 2017), <code>ggplot2</code> (Wickham 2016) and <code>ggnewscale</code> (Campitelli 2016).</p>
<p>Data available on my <a href="https://github.com/LucianoFBNeto/Tese_LucianoFBNeto">GitHub</a>.</p>
<pre class="r"><code>library(&quot;ggtree&quot;)
library(&quot;ggplot2&quot;)
library(treeio)
library(&quot;ggnewscale&quot;)
library(phytools)
library(dichromat)</code></pre>
<pre class="r"><code>#Opening and Categorizing the data
setwd(&quot;C:/Users/Luciano/Desktop/R_MarkDown&quot;)
beast_tree &lt;- read.beast(&quot;Jumbo_morfo.tree&quot;)
genotype &lt;- read.table(&quot;Data3_1.txt&quot;, header=TRUE, row.names = 1)
df2 &lt;- as.data.frame(scale(genotype[,1:18]))</code></pre>
<pre class="r"><code>#Plot tree
beast_tree</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;Jumbo_morfo.tree&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 6 tips and 5 internal nodes.
## 
## Tip labels:
## [1] &quot;Gen.N.jumbPBN&quot; &quot;Gen.N.sp.1JAG&quot; &quot;Gen.N.sp.2PIA&quot; &quot;Gen.N.sp.3IPO&quot;
## [5] &quot;Gen.N.sp.3PBM&quot; &quot;Gen.N.sp.4SFR&quot;
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;height&#39;,   &#39;height_0.95_HPD&#39;,  &#39;height_median&#39;,    &#39;height_range&#39;, &#39;length&#39;,
##  &#39;length_0.95_HPD&#39;,  &#39;length_median&#39;,    &#39;length_range&#39;, &#39;posterior&#39;,    &#39;rate&#39;,
##  &#39;rate_0.95_HPD&#39;,    &#39;rate_median&#39;,  &#39;rate_range&#39;.</code></pre>
<pre class="r"><code>p &lt;- ggtree(beast_tree, aes(x=&#39;Node_ages&#39;)) + theme_tree2() + scale_x_continuous(labels = abs) +
  geom_nodelab(aes(x=branch), vjust=-.5, size=3, color=&quot;black&quot;) + 
  geom_range(&#39;height_0.95_HPD&#39;, color=&#39;grey&#39;, alpha=.8, size=3)
p1 &lt;- revts(p)
plot(p1)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<pre class="r"><code>#Plot tips names
p2 &lt;- p1 + geom_tiplab(size=4, align=TRUE)</code></pre>
<pre><code>## Warning: Duplicated aesthetics after name standardisation: size</code></pre>
<pre class="r"><code>plot(p2)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<pre class="r"><code>#Creating heatmap with continuous data associated with the tree
gheatmap(p2, df2, offset=4.5, width=4,
         colnames=FALSE) +
  scale_fill_viridis_c(option=&quot;B&quot;, name=&quot;continuous\nvalue&quot;) + 
  scale_x_ggtree()</code></pre>
<pre><code>## Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;,
## which will replace the existing scale.</code></pre>
<pre><code>## Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
## will replace the existing scale.</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
